### PR TITLE
[libs/client] Reset cmd when a transaction gets changed

### LINKED ISSUE #107: [@kadena/client] remove `cmd` when non-malleable transaction get changed
remove `sigs` and `hash` and `cmd` when a non-malleable transactionbuilder gets changed

### PR DESCRIPTION
**Pull request by ashwinvandijk**
When a transaction gets changed after being finalized, we need to reset the hash and signatures (if applicable) because they do not match the transaction it was calculated/signed for.

Closes #107

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203499728801195

### CODE REVIEW

**Review by Randynamic — APPROVED** [review]

**Review by Randynamic — APPROVED** [review]

**Review by Randynamic — DISMISSED** [review]

File: packages/libs/client/src/tests/pact.test.ts [hunk 1]
```diff
@@ -197,4 +197,98 @@ describe('Pact proxy', () => {
 
     expect(() => builder.poll('fake-api-host.local.co')).toThrow();
   });
+
+  describe('unfinishes the transaction', () => {
```
Comment (alber70g) [inline packages/libs/client/src/tests/pact.test.ts:201]: `describe` contains the subject, `it` contains the action that is executed and the result that it should yield

I this case `describe('Pact proxy')` `it('unfinishes the transaction when setMeta is called')`

gives this sentence: `Pact proxy unfinishes the transaction when setMeta is called`

**Review by alber70g — APPROVED** [review]
Minor comment

File: CODEOWNERS [hunk 2]
```diff
@@ -2,5 +2,5 @@
 # See: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners
 
 # As a test I put myself here
-./common    @alber70g @randynamic @ggobugi27
-./*.*       @alber70g @randynamic @ggobugi27
+./common    @alber70g @randynamic @ggobugi27 @ashwinvandijk
+./*.*       @alber70g @randynamic @ggobugi27 @ashwinvandijk
```
Comment (alber70g) [inline CODEOWNERS:6]: Not sure if this is necessary. We're not working with this yet... 
The plan was to add codeowners for specific projects.
I would probably be codeowner for the `./common` as that is rush stuff

**Review by Randynamic — DISMISSED** [review]

File: packages/libs/client/src/tests/pact.test.ts [hunk 3]
```diff
@@ -198,3 +198,97 @@ describe('Pact proxy', () => {
     expect(() => builder.poll('fake-api-host.local.co')).toThrow();
   });
 });
+
+describe('The hash and signers for a transaction are reset', () => {
```
Comment (alber70g) [inline packages/libs/client/src/tests/pact.test.ts:202]:
```suggestion
describe('TransactionCommand', () => {
```

File: packages/libs/client/src/tests/pact.test.ts [hunk 4]
```diff
@@ -198,3 +198,97 @@ describe('Pact proxy', () => {
...
+      })
+      .setMeta(
+        {
+          sender: senderPubKey,
+        },
+        'testnet04',
+      );
+  });
+
+  it('when setMeta is called', async () => {
```
Comment (alber70g) [inline packages/libs/client/src/tests/pact.test.ts:240]:
```suggestion
  it('resets `cmd` `hash` and `signatures` when `setMeta` is called', async () => {
```

File: packages/libs/client/src/tests/pact.test.ts [hunk 5]
```diff
@@ -198,3 +198,97 @@ describe('Pact proxy', () => {
...
+
+    expect(updatedTransactionCommand.sigs).toEqual([undefined]);
+
+    const updatedTransaction = updatedTransactionCommand.createCommand();
+
+    expect(transaction.cmd).not.toEqual(updatedTransaction.cmd);
+    expect(transaction.hash).not.toEqual(updatedTransaction.hash);
+  });
+
+  it('when addData is called', async () => {
```
Comment (alber70g) [inline packages/libs/client/src/tests/pact.test.ts:258]:
```suggestion
  it('resets `cmd` `hash` and `signatures` when `addData` is called', async () => {
```

File: packages/libs/client/src/tests/pact.test.ts [hunk 6]
```diff
@@ -198,3 +198,97 @@ describe('Pact proxy', () => {
...
+
+    expect(updatedTransactionCommand.sigs).toEqual([undefined]);
+
+    const updatedTransaction = updatedTransactionCommand.createCommand();
+
+    expect(transaction.cmd).not.toEqual(updatedTransaction.cmd);
+    expect(transaction.hash).not.toEqual(updatedTransaction.hash);
+  });
+
+  it('when addCap is called', async () => {
```
Comment (alber70g) [inline packages/libs/client/src/tests/pact.test.ts:279]:
```suggestion
  it('resets `cmd` `hash` and `signatures` when `addCap` is called', async () => {
```

File: common/changes/@kadena/client/feat-remove-cmd-malleable_2022-12-05-12-16.json [hunk 7]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@kadena/client",
+      "comment": "Remove `cmd` from the transaction when the properties of this transaction change, so it gets recalculated. Also reset signatures (if any) in that case, because the signatures do no longer match the transaction",
```
Comment (alber70g) [inline common/changes/@kadena/client/feat-remove-cmd-malleable_2022-12-05-12-16.json:5]:
```suggestion
      "comment": "When proprerties of a non-malleable transaction change, remove `cmd`, `hash` and `signatures` from the transaction to issue a recalculation of the `cmd` and `hash`.",
```

**Review by alber70g — CHANGES_REQUESTED** [review]

**Review by alber70g — APPROVED** [review]

**Pull request merged**